temoa/_internal/table_writer.py

```
@@ -81,13 +76,64 @@ def __init__(self, config: TemoaConfig, epsilon: float = 1e-5) -> None:
         self.tech_sectors: dict[str, str] | None = None
         self.flow_register: dict[FI, dict[FlowType, float]] = {}
         self.emission_register: dict[EI, float] | None = None
+
+        # Cache for table columns to avoid repeated PRAGMA calls
+        self._table_columns_cache: dict[str, set[str]] = {}
+
         try:
             self.con = sqlite3.connect(config.output_database)
+            self.con.execute('PRAGMA foreign_keys = OFF')
         except sqlite3.OperationalError as e:
             logger.error('Failed to connect to output database: %s', config.output_database)
```

Ruff (0.14.8)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)